shpc/main/registry/remote.py

```diff
@@ -91,95 +75,106 @@ def override_exists(self, tag):
 
 
 class VersionControl(Provider):
-    def __init__(self, *args, **kwargs):
-        self.tag = kwargs.get("tag")
+
+    # The URL is substituted with owner, repo
+    library_url_schemes = {
+        "github.com": "https://%s.github.io/%s/library.json",
+        "gitlab.com": "https://%s.gitlab.io/%s/library.json",
+    }
+
+    # The URL is substituted with repo, branch, module_name
+    web_container_url_schemes = {
+        "github.com": "https://github.com/%s/blob/%s/%s/container.yaml",
+        "gitlab.com": "https://gitlab.com/%s/-/blob/%s/%s/container.yaml",
+    }
+
+    # The URL is substituted with repo, branch, module_name
+    raw_container_url_schemes = {
+        "github.com": "https://raw.githubusercontent.com/%s/%s/%s/container.yaml",
+        "gitlab.com": "https://gitlab.com/%s/-/raw/%s/%s/container.yaml",
+    }
```

Comment:
The existence of these dictionaries is perhaps the proof I should reinstate `GitHub` and `GitLab` as sub-classes of `VersionControl` ?
And instead of exposing those strings with `%s`, they would expose methods, which would give more flexibility in case a Registry provider doesn't substitute the elements in the same order

Reply:
> The existence of these dictionaries is perhaps the proof I should reinstate GitHub and GitLab as sub-classes of VersionControl ?

I like that idea I think, yes.

Reply:
Done, and I like it